src/desktop/apps/authentication/helpers.ts

```diff
@@ -103,13 +98,13 @@ export const handleSubmit = (
 
   switch (type) {
     case ModalType.login:
-      user.login(options)
+      await loginUser(userAttributes, options)
       break
     case ModalType.signup:
-      user.signup(options)
+      await signupUser(userAttributes, options)
       break
     case ModalType.forgot:
-      user.forgot(options)
+      await forgotUserPassword(userAttributes, options)
```

Comment:
Rather than passing in Promises as callbacks, I feel like the following refactoring may be easier to read:


```ts
let action = null;
switch {
  ...
  action = loginUser(userAttributes)
 ...
}

await action.then(
  options.success,
  options.error
)
```

Reply:
Tried doing something similar while assembling this, but had trouble getting access to all the async data when expected. Happy to pursue simplifying this as an improvement later.